src/main/scala/vectorpipe/VectorPipe.scala

```diff
@@ -0,0 +1,103 @@
+package vectorpipe
+
+import vectorpipe.vectortile._
+import vectorpipe.vectortile.export._
+
+import geotrellis.proj4.{CRS, LatLng}
+import geotrellis.spark.SpatialKey
+import geotrellis.spark.tiling.{ZoomedLayoutScheme, LayoutLevel}
+import geotrellis.vector.Geometry
+import geotrellis.vectortile._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql._
+import org.apache.spark.sql.functions._
+import org.locationtech.jts.{geom => jts}
+
+object VectorPipe {
+
+  case class Options(
+    maxZoom: Int,                    // Largest (most resolute) zoom level to generate
+    minZoom: Option[Int],            // Smallest (least resolute) zoom level to generate
+    srcCRS: CRS,                     // projection of the original geometry
+    destCRS: Option[CRS],            // projection for generated vectortiles
+    tileResolution: Int = 4096,      // "pixel" resolution of output vectortiles
+    geometryColumn: String = "geom"  // name of the dataframe column containing geometries
+  )
+  object Options {
+    def forZoom(zoom: Int) = Options(zoom, None, LatLng, None)
+    def forAllZoomsFrom(zoom: Int) = Options(zoom, Some(0), LatLng, None)
+    def forAllZoomsWithSrcProjection(zoom: Int, crs: CRS) = Options(zoom, Some(0), crs, None)
+  }
+
+  def apply(input: DataFrame, pipeline: vectortile.Pipeline, layerName: String, options: Options): Unit = {
```

Comment:
Thoughts on breaking a DataFrame of features into multiple layers in a vector tileset?

Reply:
Probably a good idea.  I don't exactly want to ask a user to manually interact with the `geotrellis.vectortile` interface, as I think it's weird and kinda crappy.  There might be a way to provide a `Seq[String]` as `layerNames` and pass the layer name to the `select` function?  It's not a great way to go, but would give some ability to separate things out.

Relatedly, do you think it would be beneficial to allow the specification of the version number of the vector tiles?  I'm not even sure if GT can output anything but V2 tiles ...

Reply:
In practice, I think people are extremely unlikely to need to modify the MVT version in use (especially since v2 is in widespread use at this point). Modifying the tile resolution is also pretty unlikely, but there are probably good reasons (I don't know of any, but hey).

Reply:
Re: layers and layer names, see below. The short version is that I think `layerName` should result from `Pipeline` methods, not be provided as an argument to `VectorPipe`.

Reply:
Proposed solution in 165637d80ad72f1b57112dab973bc8648bb2ac31

Reply:
Consolidating this and other conversations down to https://github.com/geotrellis/vectorpipe/pull/67#discussion_r273734616